text/0000-raw-keywords.md

```diff
@@ -0,0 +1,270 @@
+A few examples of what this can include are:
+
+- For language, library, cargo, tools, and compiler proposals: Does this feature exist in other programming languages and what experience have their community had?
+- For community proposals: Is this done by some other community and what were their experiences with it?
+- For other teams: What lessons can we learn from what other communities have done here?
+- Papers: Are there any published papers or great posts that discuss this? If you have some relevant papers to refer to, this can serve as a more detailed theoretical background.
+
+This section is intended to encourage you as an author to think about the lessons from other languages, provide readers of your RFC with a fuller picture.
+If there is no prior art, that is fine - your ideas are interesting to us whether they are brand new or if it is an adaptation from other languages.
+
+Note that while precedent set by other languages is some motivation, it does not on its own motivate an RFC.
+Please also take into consideration that rust sometimes intentionally diverges from common language features.
+-->
+
+This is patterned on [RFC #2151, `raw_identifiers`](https://rust-lang.github.io/rfcs/2151-raw-identifiers.html).
+
+Some scripting languages take the opposite approach and essentially reserve all unprefixed identifiers as keywords, requiring a sigil (such as `$foo`) to have it be interpreted as an identifier in an expression.  This is clearly infeasible for rust, due to the extraordinary churn it would require.
+
+C reserves all identifiers starting with an underscore, and uses that along with `#define` to add features.  For example, it added `_Bool`, and made that available as `bool` only when `#include <stdbool.h>` is specified.  Rust doesn't need this for types (as `i32` and friends are not keywords), but could add new syntax constructs as macros.
+
+C# releases new versions [irregularly](https://en.wikipedia.org/wiki/C_Sharp_%28programming_language%29#Versions), major versions of which may include source-breaking changes such as new keywords.  Rust could decide to just roll editions more often instead of introducing features in the middle of them.
+
+C# also leverages contextual keywords heavily.  For example, `await` is only a keyword inside functions using the `async` contextual keyword, so they could be introduced as non-breaking.  This kind of contextual behaviour is more awkward for rust, which needs to be able to parse an `expr` to pass it to a macro.
+
+Python uses [*future statements*](https://docs.python.org/3/reference/simple_stmts.html#future) to allow use of the new features on a per-module basis before those feature become standard.  Rust's `#![feature(foo)]` on nightly is similar here.
+
+Haskell has the [`LANGUAGE` pragma](https://ghc.readthedocs.io/en/8.0.2/glasgow_exts.html#language-pragma), which `ghc` also supports as command line parameters.  This is again similar to Rust's `#![feature(foo)]` on nightly.
```

Comment:
Would you mind saying more why using `#![feature]` to enable new keywords (on stable) isn't possible or desirable? I believe that the parser in rustc currently does not know about features, but I do not see why it couldn't (except maybe complexity with macros).